my-react-app/src/redux/login/reducer.js

```diff
@@ -0,0 +1,33 @@
+import { actions } from './actions';
+
+const initialState = {
+  token: localStorage.getItem('token'),
+  auth: !!localStorage.getItem('auth'),
+  emailInvalid: false,
+  passwordInvalid: false
+};
+
+export function reducerLogin(state = initialState, action) {
+  switch (action.type) {
+    case actions.GET_TOKEN:
+      return {
+        state
+      };
+    case actions.GET_TOKEN_SUCCESS:
+      localStorage.setItem('auth', action.payload.auth);
```

Comment:
You shouldn't have side effects in a reducer, except for manipulating the redux store.